src/repository.py

```diff
@@ -34,6 +41,47 @@
 CONFIG_USER_EMAIL = (CONFIG_USER_SECTION_NAME, "email")
 
 
+class DiffSummary(NamedTuple):
+    is_dirty: bool
+    new: Sequence[str]
+    removed: Sequence[str]
+    modified: Sequence[str]
+
+    @classmethod
+    def from_raw_diff(cls, diffs: Sequence[Diff]):
+        new_files = {diff.a_path for diff in diffs if diff.new_file}
+        removed_files = {diff.a_path for diff in diffs if diff.deleted_file}
+        modified_files = {diff.a_path for diff in diffs if diff.renamed_file}.union(
```

Comment:
Could this be `{diff.a_path for diff in diffs if diff.renamed_file or diff.change_type == "M"}`?